### ISSUE TITLE
shift in byte mode should produce syntax error

### ISSUE THREAD

**Issue by arcanist123** [abaplint/transpiler#420]
This should not be possible to transpile - the lv_temp type should not allow shift in byte mode

``` abap 
  DATA lv_temp TYPE string.
  lv_temp = 'abc'.
  SHIFT lv_temp BY 1 PLACES LEFT in byte mode.
  WRITE lv_temp.
```

**Comment by larshp** [comment]
a few notes on syntax errors:

it is the responsibility of [abaplint](https://abaplint.org) to ensure the syntax is correct, there is also a lot of work to be done here

the transpiler takes the AST, and assumes its correct ABAP syntax and does it best to transpile it

**Comment by arcanist123** [comment]
Agreed. I opened a separate issue in abaplint repository.
This code should fail with runtime error instead of executing without any issue 
``` abap
  DATA lv_temp TYPE string.
  lv_temp = 'abc'.
  FIELD-SYMBOLS <lv> TYPE any.
  ASSIGN lv_temp TO <lv>.
  SHIFT <lv> BY 1 PLACES LEFT IN BYTE MODE.
  WRITE lv_temp.
```

**Comment by larshp** [comment]
runtime error = OBJECTS_NOT_BYTELIKE
